### PR TITLE
Drop Vatlayer

### PR DESCRIPTION
**Pull request by maarcingebala**
This PR removes Vatlayer plugin, as its functionalities will be replaced by tax classes.
- If Vatlayer is enabled, we create new tax classes based on Vatlayer's tax codes ones used in products and product types; we also copy VAT rates, including custom rules `countries_to_calculate_taxes_from_origin` and `excluded_countries` (see original Vatlayer plugin config).
- Deprecated `shopFetchTaxRates` mutation.
- `django_prices_vatlayer` package will be removed in a separate PR, as we still use `VAT` model in some places.

**Breaking changes**
This PR removes two plugin/manager methods, as they are no longer used:
- `fetch_taxes_data`,
- `get_tax_rate_percentage_value`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6796 Drop Vatlayer](https://app.clickup.com/t/2549495/SALEOR-6796)

**Comment by patrys** [comment]
Task linked: [SALEOR-6391 Simple taxes](https://app.clickup.com/t/2549495/SALEOR-6391)

File: saleor/tax/migrations/0004_migrate_vatlayer.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        origin_country = _clear_country_code(config_dict["origin_country"])
+        countries_to_calculate_taxes_from_origin = _clear_str_list_country_codes(
+            config_dict["countries_to_calculate_taxes_from_origin"]
+        )
+        for country in countries_to_calculate_taxes_from_origin:
+            use_origin_country_map[country] = origin_country
+
+        # Migrate `excluded_countries` to new tax configuration.
+        excluded_countries = _clear_str_list_country_codes(
+            config_dict["excluded_countries"]
```
Comment (korycins) [inline saleor/tax/migrations/0004_migrate_vatlayer.py:110]: Assuming that you already check that, but I would propose to double check if the `config_dict` have always added fields like `excluded_countries`

Comment (korycins) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:110]: Especially when you're migrating from version like 2.11 where you had active vatlayer (this version didn't have fields like `excluded_countries`).

File: saleor/tax/migrations/0004_migrate_vatlayer.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                rates[(tax_class.id, vat.country_code)] = reduced_rate
+
+        # Swap rates for countries that should use origin country tax rate instead of
+        # own rates.
+        for country_code, origin in use_origin_country_map.items():
+            country_rate_obj = rates.get((tax_class.id, country_code))
+            origin_rate_obj = rates.get((tax_class.id, origin))
+            if country_rate_obj and origin_rate_obj:
+                country_rate_obj.rate = origin_rate_obj.rate
+                rates[(tax_class.id, country_code)] = country_rate_obj
```
Comment (korycins) [inline saleor/tax/migrations/0004_migrate_vatlayer.py:161]: I don't think that you need to re-assign it here

**Review by korycins — COMMENTED** [review]
Good job Marcin! 
You're dropping one of the first features that I've added to Saleor 🥲

File: saleor/tax/migrations/0004_migrate_vatlayer.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+def migrate_vatlayer(apps, _schema_editor):
+    PluginConfiguration = apps.get_model("plugins", "PluginConfiguration")
+
+    vatlayer_configs = PluginConfiguration.objects.filter(
+        active=True, identifier=VATLAYER_ID
+    )
+    is_vatlayer_enabled = vatlayer_configs.exists()
+
+    create_tax_classes(apps)
```
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:174]: Yes, it's OK if we import them, it doesn't enable taxes for you unexpectedly; we just recreate data that you had configured in your products anyway.

File: saleor/tax/migrations/0004_migrate_vatlayer.py (hunk 2 above)
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:161]: I'm reassigning to override the original tax rate of a country with the rate from the "origin" country.

File: saleor/tax/migrations/0004_migrate_vatlayer.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:110]: Good point.

File: saleor/tax/migrations/0004_migrate_vatlayer.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
+# Generated by Django 3.2.14 on 2022-07-27 16:05
+
+from collections import defaultdict
+from typing import List, Optional
+from django_countries import countries
+from django.db import migrations
+from django_prices_vatlayer.models import VAT
```
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:7]: Do you mean to use apps.get_model?

File: saleor/tax/migrations/0004_migrate_vatlayer.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    use_origin_country_map = {}
+
+    for vatlayer_config in vatlayer_configs:
+        config_dict = {
+            item["name"]: item["value"] for item in vatlayer_config.configuration
+        }
+        channel = vatlayer_config.channel
+
+        # Migrate `countries_to_calculate_taxes_from_origin` setting.
+        origin_country = _clear_country_code(config_dict["origin_country"])
```
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:101]: In the `migrate_vatlayer function I only fetch the active ones.

File: saleor/tax/migrations/0004_migrate_vatlayer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:174]: Yeah, but If I have such data and I don't have any active vatlayer channel, it means that I don't use it. In that case why do we want to make a "mess" and create a tax classes that are not used?

File: saleor/tax/migrations/0004_migrate_vatlayer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:7]: Yes. Not sure if this works with external modules but I definitely think that we should not import models directly in the migration. (Especially, with this line, you're not able to drop the package `djago_prices_vatlayer`).

File: saleor/tax/migrations/0004_migrate_vatlayer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py:175]: You have it anyway in your metadata and we cannot guess if you need it or not. I assume we can migrate it for you and if you don't need it, you will remove it. If you does need it and we don't migrate it, you have to recreate a lot of stuff.

File: saleor/tax/migrations/0004_migrate_vatlayer.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_vatlayer.py]: Fixed

File: saleor/plugins/base_plugin.py [hunk 6]
```diff
@@ -468,9 +468,6 @@ def __str__(self):
         Tuple[Union["User", NoneType], dict],
     ]
 
-    #  Triggered when ShopFetchTaxRates mutation is called.
-    fetch_taxes_data: Callable[[Any], Any]
-
```
Comment (fowczarek) [inline saleor/plugins/base_plugin.py:473]: Should those changes be described in the changelog as breaking changes?

**Review by fowczarek — APPROVED** [review]

File: saleor/plugins/base_plugin.py (hunk 6 above)
Comment (maarcingebala) [reply saleor/plugins/base_plugin.py:473]: Yes, it will be added to changelog later when merging the feature branch to `main`.

**Review by korycins — APPROVED** [review]

**Pull request merged**